packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2

```diff
@@ -697,20 +703,19 @@ def _get_deserialize_callable_from_annotation(  # pylint: disable=R0911, R0915,
     try:
         if any(a for a in annotation.__args__ if a == type(None)):  # pyright: ignore
             if_obj_deserializer = _get_deserialize_callable_from_annotation(
-                next(a for a in annotation.__args__ if a != type(None)), module, rf  # pyright: ignore
+                _sorted_annotations(annotation.__args__)[0], module, rf  # pyright: ignore
```

Comment:
It should be a legacy bug here. The optional type could also be a union. I suggest you to filter out the none type and split into two if:
1. only one type left, keep current logic
2. more than one type, construct a new union annotation and call the `_get_deserialize_callable_from_annotation`

Reply:
agreed